cli/command/image/prune_test.go

```diff
@@ -0,0 +1,121 @@
+	testCases := []struct {
+		name            string
+		args            []string
+		imagesPruneFunc func(pruneFilter filters.Args) (types.ImagesPruneReport, error)
+	}{
+		{
+			name: "wrong-args",
+			args: []string{"something"},
+		},
+		{
+			name: "prune-error",
+			args: []string{"--force"},
+			imagesPruneFunc: func(pruneFilter filters.Args) (types.ImagesPruneReport, error) {
+				return types.ImagesPruneReport{}, errors.Errorf("somethign went wrong")
+			},
+		},
+	}
+	for _, tc := range testCases {
+		buf := new(bytes.Buffer)
+		cmd := NewPruneCommand(test.NewFakeCli(&fakeClient{
+			imagesPruneFunc: tc.imagesPruneFunc,
+		}, buf))
+		cmd.SetOutput(ioutil.Discard)
+		cmd.SetArgs(tc.args)
+		err := cmd.Execute()
+		assert.NotNil(t, err)
+		actual := err.Error()
+		expected := string(golden.Get(t, []byte(actual), fmt.Sprintf("prune-command-errors.%s.golden", tc.name))[:])
+		assert.EqualNormalizedString(t, assert.RemoveSpace, actual, expected)
+	}
+}
+
+func TestNewPruneCommandSuccess(t *testing.T) {
+	testCases := []struct {
+		name            string
+		args            []string
+		imagesPruneFunc func(pruneFilter filters.Args) (types.ImagesPruneReport, error)
+	}{
+		{
+			name: "all",
+			args: []string{"--all"},
+			imagesPruneFunc: func(pruneFilter filters.Args) (types.ImagesPruneReport, error) {
+				assert.Equal(t, pruneFilter.Get("dangling")[0], "false")
+				return types.ImagesPruneReport{}, nil
+			},
+		},
+		{
+			name: "force-deleted",
+			args: []string{"--force"},
+			imagesPruneFunc: func(pruneFilter filters.Args) (types.ImagesPruneReport, error) {
+				assert.Equal(t, pruneFilter.Get("dangling")[0], "true")
+				return types.ImagesPruneReport{
+					ImagesDeleted:  []types.ImageDeleteResponseItem{{Deleted: "image1"}},
+					SpaceReclaimed: 1,
+				}, nil
+			},
+		},
+		{
+			name: "force-untagged",
+			args: []string{"--force"},
+			imagesPruneFunc: func(pruneFilter filters.Args) (types.ImagesPruneReport, error) {
+				assert.Equal(t, pruneFilter.Get("dangling")[0], "true")
+				return types.ImagesPruneReport{
+					ImagesDeleted:  []types.ImageDeleteResponseItem{{Untagged: "image1"}},
+					SpaceReclaimed: 2,
+				}, nil
+			},
+		},
+	}
+	for _, tc := range testCases {
+		buf := new(bytes.Buffer)
+		cmd := NewPruneCommand(test.NewFakeCli(&fakeClient{
+			imagesPruneFunc: tc.imagesPruneFunc,
+		}, buf))
+		cmd.SetOutput(ioutil.Discard)
+		cmd.SetArgs(tc.args)
+		err := cmd.Execute()
+		assert.NilError(t, err)
+		actual := buf.String()
+		expected := string(golden.Get(t, []byte(actual), fmt.Sprintf("prune-command-success.%s.golden", tc.name))[:])
+		assert.EqualNormalizedString(t, assert.RemoveSpace, actual, expected)
+	}
+}
+
+func TestRunPruneMethodCall(t *testing.T) {
+	cli := test.NewFakeCli(&fakeClient{}, new(bytes.Buffer))
+	filter := opts.NewFilterOpt()
+	filter.Set("filter=false")
+	functionCalled := false
+	runPrune = func(dockerCli command.Cli, opts pruneOptions) (spaceReclaimed uint64, output string, err error) {
```

Comment:
This test doesn't really test anything except of a function call that is tested by exercising the code in previous test, so it's not needed.